sql/plan/show_function_status.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2022 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plan
+
+import (
+	"strings"
+
+	"github.com/dolthub/go-mysql-server/sql"
+)
+
+type ShowFunctionStatus struct {
+	RoutinesTable sql.Node
+}
+
+var _ sql.Node = (*ShowFunctionStatus)(nil)
+
+var showFunctionStatusSchema = sql.Schema{
+	&sql.Column{Name: "Db", Type: sql.LongText, Nullable: false},
+	&sql.Column{Name: "Name", Type: sql.LongText, Nullable: false},
+	&sql.Column{Name: "Type", Type: sql.LongText, Nullable: false},
+	&sql.Column{Name: "Definer", Type: sql.LongText, Nullable: false},
+	&sql.Column{Name: "Modified", Type: sql.Datetime, Nullable: false},
+	&sql.Column{Name: "Created", Type: sql.Datetime, Nullable: false},
+	&sql.Column{Name: "Security_type", Type: sql.LongText, Nullable: false},
+	&sql.Column{Name: "Comment", Type: sql.LongText, Nullable: false},
+	&sql.Column{Name: "character_set_client", Type: sql.LongText, Nullable: false},
+	&sql.Column{Name: "collation_connection", Type: sql.LongText, Nullable: false},
+	&sql.Column{Name: "Database Collation", Type: sql.LongText, Nullable: false},
+}
+
+// NewShowFunctionStatus creates a new *ShowFunctionStatus node.
+func NewShowFunctionStatus() *ShowFunctionStatus {
+	return &ShowFunctionStatus{}
+}
+
+// String implements the sql.Node interface.
+func (s *ShowFunctionStatus) String() string {
+	return "SHOW PROCEDURE STATUS"
+}
+
+// Resolved implements the sql.Node interface.
+func (s *ShowFunctionStatus) Resolved() bool {
+	return true
+}
+
+// Children implements the sql.Node interface.
+func (s *ShowFunctionStatus) Children() []sql.Node {
+	if s.RoutinesTable == nil {
```

Comment:
When is this ever nil?

Reply:
If this is a copied pattern then its fine. I'd rather you return an empty list not nil